recommender/static/js/src/recommender.js

```diff
@@ -1,1022 +1,1187 @@
+                Logger.log('mit.recommender.editResource', generateLog(loggerStatus['editResource']['complete'], information));
+
+                /* Edit resource when the screenshot isn't/is provided */
+                if ($("input[name='file']", formDiv).val() === '') { editResource(data); }
+                else { writeResourceWithScreenshot(formDiv, file, writeDatabaseEnum.EDIT, data); }
+            });
+        });
+    }
+
+    /** 
+     * Bind the event for flagging problematic resource and submitting the
+     * reason why student think the resource is problematic.
+     * @param {element} ele The recommenderResource element the flag event will be bound to.
+     */
+    function bindResourceFlagEvent(ele) {
+        $('.flagResource', ele).click(function() {
+            showModifyingPage('.flagResourcePage')
+
+            var flagDiv = $(this);
+            var flaggedResourceDiv = $(this).parent().parent();
+            $('.flagReason', element).val($('.recommenderProblematicReason', flaggedResourceDiv).text());
+            data = {};
+            data['id'] = $('.recommenderEntryId', flaggedResourceDiv).text();
+            Logger.log('mit.recommender.flagResource', generateLog(loggerStatus['flagResource']['attempt'], {'id': data['id']}));
+
+            $('.flagReasonSubmit', element).unbind();
+            $('.unflagButton', element).unbind();
+
+            /* Flag the problematic resource and save the reason to database */ 
+            $('.flagReasonSubmit', element).click(function() {
+                data['reason'] = $('.flagReason', element).val();
+                data['isProblematic'] = true;
+                Logger.log('mit.recommender.flagResource', generateLog(loggerStatus['flagResource']['complete'], data));
+
+                $.ajax({
+                    type: "POST",
+                    url: flagResourceUrl,
+                    data: JSON.stringify(data),
+                    success: function(result) {
+                        var flaggedResourceDiv = $('.recommenderResource:eq(' + findResourceDiv(result['id']).toString() + ')', element);
+                        var flagDiv = $('.flagResource:eq(' + findResourceDiv(result['id']).toString() + ')', element);
+        
+                        $('.recommenderProblematicReason', flaggedResourceDiv).text(result['reason']);
+                        if (result['isProblematic']) { $(flagDiv).addClass('problematic'); }
+                        else { $(flagDiv).removeClass('problematic'); }
+                        addResourceDependentTooltip(flaggedResourceDiv);
+                        setFlagAriaParam(flaggedResourceDiv);
+                        backToView();
+                    }
+                });
+            });
+
+            /* Unflag the resource */
+            $('.unflagButton', element).click(function() {
+                data['isProblematic'] = false;
+                Logger.log('mit.recommend
```

Comment:
This seems like a perfect use for the [Underscore](http://underscorejs.org/) library -- specifically, [`_.find`](http://underscorejs.org/#find). There's no need to refactor all your Javascript to use Underscore, so this is just a nitpick, but it's a useful library and one that you should be aware of.